File: ibm-connnections/GetUserInformationOnProfiles.py
# -*- coding: utf-8 -*-
#
# Antes de executar:
#
# > pip install requests
#
#
import sys
import requests
from requests.packages.urllib3.exceptions import InsecureRequestWarning
from requests.auth import HTTPBasicAuth
import xml.etree.ElementTree as ET

CONNECTIONS_HOST = 'https://connections.<REPLACE_HERE>.com'
CONNECTIONS_USERNAME = '<REPLACE_HERE>'
CONNECTIONS_PASSWORD = '<REPLACE_HERE>'
QUERY = 'email=smith@<REPLACE_HERE>.com'
# Other options for QUERY
# 1) email=<email_address>
# 2) uuid=<uuid of user>
# 3) uid=<uid of user>

DEBUG=0

# Disable Warnings from Untrusted TLs keys
requests.packages.urllib3.disable_warnings(InsecureRequestWarning)

# Disable Warnings from Untrusted TLs keys
def doGet(url,headers,auth):
    res = requests.get(url=url,headers=headers,auth=auth, verify=False)
    if (res.status_code != 200):
        print 'requests.get -> %s = %s\n' % (res.url, res)
        return None;
    #print res.content
    return res.content

def parseProfileIfno(content):

    root = ET.fromstring(content)
    entry = root.find("{http://www.w3.org/2005/Atom}entry")
    contributor = entry.find('{http://www.w3.org/2005/Atom}contributor')
    name = contributor.find('{http://www.w3.org/2005/Atom}name').text
    # for namespace snx:userid change to http://www.ibm.com/xmlns/prod/sn
    userid = contributor.find('{http://www.ibm.com/xmlns/prod/sn}userid').text
    email = contributor.find('{http://www.w3.org/2005/Atom}email').text
    userState = contributor.find('{http://www.ibm.com/xmlns/prod/sn}userState').text

    profile = {
    'name': name,
    'userid': userid,
    'email': email,
    'userState': userState
    }
    #print profile
    return profile

def getProfileInfo(query):
    url = CONNECTIONS_HOST + '/profiles/atom/profile.do?' + query
    print url + '\n'
    headers = { 'Content-Type': 'application/atom+xml;charset=UTF-8'}
    auth=None

    feed = doGet(url=url,headers=headers,auth=auth)
    if (feed is None):
        return None
    return parseProfileIfno(feed)

#################### Main Module ###################
print 'Connecting to IBM Connections...\n'
if (DEBUG == 1):
    print sys.stdout.encoding + '\n\n'

profile = getProfileInfo(QUERY)

if (profile is None):
    print 'Cannot get Profile Information.'
    sys.exit(1)

print 'Profile Information:\n'
print profile['name']
print ' |--> userid: ' + profile['userid']
print ' |--> email: ' + profile['email']
print ' |--> userState: ' + profile['userState']

File: ibm-connections/GetUserInformationOnProfiles.js
/*
 * Before Execute
 *
 * > npm install request
 * > npm install xml2js
 *
 *
 */
'use strict';
// Ignore self signed certificates
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

const request = require('request');
const xml2js = require('xml2js');
const parser = new xml2js.Parser();

const CONNECTIONS_HOST = 'https://connections.<REPLACE_HERE>.com';
const CONNECTIONS_USERNAME = '<REPLACE_HERE>';
const CONNECTIONS_PASSWORD = '<REPLACE_HERE>';
const QUERY = 'email=smith@<REPLACE_HERE>.com';
/*
 * Other options for QUERY
 *  1) email=<email_address>
 * 2) uuid=<uuid of user>
 * 3) uid=<uid of user>
 */

function getProfileInfo() {
    var options = {
        url: CONNECTIONS_HOST + '/profiles/atom/profile.do?' + QUERY,
        auth: {
            username: CONNECTIONS_USERNAME,
            password: CONNECTIONS_PASSWORD
        },
        headers: {
            'Content-Type': 'application/atom+xml;charset=UTF-8'
        }
    };

    request.get(options, function(error, response, body) {
        if (error || response.statusCode != 200) {
            console.error('Failed to contact Connections Server\n');
            console.error(error);
            return;
        }

        var profile = parseProfileInfo(body);
        if (!profile) {
            return;
        }
        printProfile(profile);
    });
}

function parseProfileInfo(body) {
    var profile = {};
    parser.parseString(body, function(err, result) {
        if (err) {
            console.log(err);
            console.error('parseProfileInfo: Cannot parse ProfileInfo ');
            return;
        }
        if (!result.feed.entry) {
            console.error('parseProfileInfo: Missing <ENTRY> element');
            return;
        }

        profile['name'] = result.feed.entry[0].contributor[0]['name'][0];
        profile['email'] = result.feed.entry[0].contributor[0]['email'][0];
        profile['userid'] = result.feed.entry[0].contributor[0]['snx:userid'][0];
        profile['userState'] = result.feed.entry[0].contributor[0]['snx:userState'][0];
    });
    return profile;
}

function printProfile(profile) {
    console.log('Profile Information:');
    console.log(profile['name']);
    console.log(' |--> userid: ' + profile['userid']);
    console.log(' |--> email: ' + profile['email']);
    console.log(' |--> userState: ' + profile['userState']);
}

//******************************     MAIN          **********************************//
console.log('Connecting to IBM Connections...\n');
getProfileInfo();
